app/models/manager_refresh/inventory_collection.rb

```diff
@@ -52,40 +57,16 @@ def to_hash
 
     def process_strategy(strategy_name)
       case strategy_name
-      when :local_db_cache_all, :local_db_find_one
-        send("process_strategy_#{strategy_name}")
-      when :find_missing_in_local_db
+      when :local_db_cache_all
+        self.data_collection_finalized = true
```

Comment:
this is more a flag for 'no more references are coming' so we can do the query and cache it